experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -0,0 +1,619 @@
+    end
+    return D
+end
+
+function reynolds_v_slm(elem::MPolyDecRingElem, new_rep_mat::AbstractAlgebra.Generic.MatSpaceElem, new_det::MPolyDecRingElem, m::Int)
+    D = mu_star(new_rep_mat)
+    mixed_ring_xy = parent(elem)
+    sum_ = mixed_ring_xy()
+    #mu_star: 
+    mons = collect(monomials(elem))
+    coeffs = collect(coefficients(elem))
+    for i in 1:length(mons)
+        k = mixed_ring_xy(1)
+        factors = factorise(mons[i])
+        for j in 1:length(factors)
+            if factors[j][2] != 0
+                neww = getindex(D, factors[j][1])
+                k = k*(neww^(factors[j][2]))
+            end
+        end
+        sum_ += coeffs[i]*k
+    end
+    t = needed_degree(sum_, m)
+    if !divides(t, m)[1]
+        return parent(elem)()
+    else
+        p = divexact(t, m)
+    end
+    #num = omegap_(p, new_det, sum_)
+    #den = omegap_(p, new_det, (new_det)^p)
+    #if !(denominator(num//den)==1)
+    #    error("denominator of reynolds not rational")
+    #end
+    #return numerator(num//den)
+    return reynolds_slm(sum_, new_det, p)
+end
+
+function reynolds_slm(elem::MPolyElem, det_::MPolyElem, p::Int)
+    num = omegap_(p,det_, elem)
+    den = omegap_(p,det_,det_^p)
+    if !(denominator(num//den)==1)
+        error("denominator of reynolds not rational")
+    end
+    return numerator(num//den)
+end
+
+function needed_degree(elem::MPolyDecRingElem, m::Int)
+    R = parent(elem)
+    n = numerator((ngens(R) - m^2)//2)
+    extra_ring, _= PolynomialRing(base_ring(R), "z"=>1:m^2)
+    mapp = hom(R,extra_ring, vcat([1 for i in 1:2*n], gens(extra_ring)))
+    return total_degree(mapp(elem))
+end
+
+ function omegap_(p::Int, det_::MPolyDecRingElem, f::MPolyDecRingElem)
+     parent(det_) == parent(f) || error("Omega process ring error")
+     action_ring = parent(det_)
+     monos = collect(monomials(det_))
+     coeffs = collect(coefficients(det_))
+     for i in 1:p
+     h = action_ring()
+     for i in 1:length(monos)
+         exp_vect = exponent_vector(monos[i], 1)
+         x = f
+         for i in 1:length(exp_vect)
+             for j in 1:exp_vect[i]
+                 x = derivative(x, i)
+                 if x == 0
+                     break
+                 end
+             end
+         end
+         h += coeffs[i]*x
+     end
+         f = h
+     end
+     return f
+ end
+
+#####################callable reynold's operator
+
+function reynolds_operator(elem::MPolyElem, X::RepresentationReductiveGroup)
+    X.group.group[1] == :SL || error("Only implemented for SLm")
+    vector_ring = parent(elem)
+    G = X.group
+    n = Int(ngens(vector_ring))
+    n == ncols(X.rep_mat) || error("group not compatible with element")
+    m = G.group[2]
+    R, _ = grade(PolynomialRing(G.field,"x"=>1:n, "y"=>1:n, "z"=>(1:m, 1:m))[1])
+    map1 = hom(vector_ring, R, gens(R)[1:n])
+    new_elem = map1(elem)
+    group_ring = base_ring(X.rep_mat)
+    map = hom(group_ring, R, gens(R)[2*n+1:2*n+m^2])
```

Comment:
I would advise against using `map` as name here, as that shadows the name of a well-known Julia function. Maybe `f` or `\phi` or so?